### PR TITLE
Chore: How to add to Storybook addon catalog documentation.

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the addon documentation is updated to include information on how to add to the addon catalog.

What was done:
- Created the documentation
- Updated the toc file
- Included assets

Feel free to provide feedback

### CODE REVIEW

File: docs/addons/addon-catalog.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+</div>
+
+### Module metadata
+
+The items mentioned above are the baseline requirements for any Storybook addon. But we rely on key pieces of information to feature the addon in the catalog:
+
+| Property    | Description                            | Example                                                               |
+| ----------- | -------------------------------------- | --------------------------------------------------------------------- |
+| name        | Addon display name                     | My-addon                                                              |
```
Comment (shilman) [inline docs/addons/addon-catalog.md:28]:
```suggestion
| name        | Addon package name                     | my-addon                                                             |
```

File: docs/addons/addon-catalog.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+To get the addon listed in the catalog, add the <code>storybook-addons</code> as the first keyword, followed by an entry to categorize the addon. Including additional keywords helps with searchability. 
+</div>
+
+### Catalog metadata
+
+Along with module related information, you'll need to specify additional metadata. In your `package.json` add the `storybook` property and use the following fields:
+
+| Property              | Description                    | Example           |
+| --------------------- | ------------------------------ | ----------------- |
+| displayName           | Addon display name             | My-addon          |
```
Comment (shilman) [inline docs/addons/addon-catalog.md:44]:
```suggestion
| displayName           | Addon display name             | My Addon          |
```

File: docs/addons/addon-catalog.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+| icon                  | Custom icon for the addon      | my-addon-icon.png |
+| unsupportedFrameworks | List of unsupported frameworks | Vue               |
+| supportedFrameworks   | List of supported frameworks   | React,Angular     |
+
+For instance:
+
+```json
+{
+  "storybook": {
+    "displayName": "My-addon",
```
Comment (shilman) [inline docs/addons/addon-catalog.md:54]:
```suggestion
    "displayName": "My Addon",
```

**Review by shilman — CHANGES_REQUESTED** [review]
LGTM @jonniebigodes with small changes

**Review by winkerVSbecks — APPROVED** [review]

**Comment by domyen** [comment]
I updated the PR for accuracy and clarity. Please review.

**Review by domyen — APPROVED** [review]

**Comment by jonniebigodes** [comment]
Self merging this.

**Pull request merged**